src/components/AvatarRow/AvatarRow.jsx

```diff
@@ -0,0 +1,172 @@
+
+function AvatarRow({
+  adaptable = true,
+  avatars = [],
+  className,
+  'data-cy': dataCy = 'AvatarRow',
+  extraTooltipProps = {},
+  gap = 2,
+  ieCompatible = false,
+  throttleOnResize = true,
+  throttleWait = 200,
+  ...avatarProps
+}) {
+  const { size: avatarSize, fontSize: avatarFontSize } = avatarConfigSizes[
+    avatarProps.size || 'md'
+  ]
+  const avatarRowRef = useRef(null)
+  const [numberOfItemsOnDisplay, setNumberOfItemsOnDisplay] = useState(
+    avatars.length
+  )
+
+  const throttledOnResize = !ieCompatible
+    ? throttle(onResize, throttleWait)
+    : noop
+  const throttledHandleWindowResize = ieCompatible
+    ? throttle(handleWindowResize, throttleWait)
+    : noop
+
+  useEffect(() => {
+    if (!adaptable || ieCompatible || avatarRowRef.current == null) return
+
+    const avatarRowEl = avatarRowRef.current
+    const resizeObserver = setupObserver(
+      throttleOnResize ? throttledOnResize : onResize
+    )
+
+    resizeObserver.observe(avatarRowEl)
+
+    return () => {
+      if (!adaptable || ieCompatible || avatarRowEl == null) return
+
+      resizeObserver.unobserve(avatarRowEl)
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [])
+
+  useEffect(() => {
+    if (ieCompatible && adaptable) {
+      window.addEventListener(
+        'resize',
+        throttleOnResize ? throttledHandleWindowResize : handleWindowResize
+      )
+
+      const measures = avatarRowRef.current.getBoundingClientRect()
+      onResize(measures)
+    }
+
+    return () => {
+      ieCompatible &&
+        adaptable &&
+        window.addEventListener(
+          'resize',
+          throttleOnResize ? throttledHandleWindowResize : handleWindowResize
+        )
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [])
+
+  function handleWindowResize() {
+    if (avatarRowRef.current != null) {
+      const measures = avatarRowRef.current.getBoundingClientRect()
+
+      onResize(measures)
+    }
+  }
+
+  function onResize({ width: containerWidth }) {
+    /** Only act if we have more than 1 avatar */
+    if (!containerWidth || avatars.length <= 1) {
+      return
+    }
+
+    /** The total space for the avatars is comprised of:
+     * Avatar space: Number of avatars * the size of the avatar
+     * +
+     * Margin space: gap between avatars * the number of gaps. For example, for 3 avatars, there are 2 gaps => [AV]gap[AV]gap[AV]
+     */
+    const spaceForAllAvatars =
+      avatarSize * avatars.length + (avatars.length - 1) * gap
+
+    if (containerWidth >= spaceForAllAvatars) {
+      setNumberOfItemsOnDisplay(avatars.length)
+    } else {
+      const numberOfGaps = numberOfItemsOnDisplay - 1
+      const itemsThatFit = Math.floor(
+        (containerWidth - numberOfGaps * gap) / avatarSize
+      )
+
+      setNumberOfItemsOnDisplay(itemsThatFit > 0 ? itemsThatFit : 1)
+    }
+  }
+
+  const { shownAvatars, hiddenAvatars } = splitAvatarsArray(
```

Comment:
I do wonder if we should wrap that with useMemo and update it only if `numberOfItemsOnDisplay` and/or the avatars value change (might be an issue if the parent doesn't memoize the avatar list). I just don't like having to split the avatars array on every call.

Reply:
In this case I didn't add it because the splitting of the avatars is only a couple of `slice` operations which I believe (but have not benchmarked) are very fast, so I'm not sure if the overhead of memoizing this is larger than the slicing.